plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -455,8 +455,16 @@ public long deleteDocuments(RemoteTableName remoteTableName, TupleDomain<ColumnH
     public MongoCursor<Document> execute(MongoTableHandle tableHandle, List<MongoColumnHandle> columns)
     {
         Document output = new Document();
-        for (MongoColumnHandle column : columns) {
-            output.append(column.getName(), 1);
+
+        if (tableHandle.getProjectedColumns().isPresent()) {
```

Comment:
Why do we need this condition instead of just relying on `columns` passed by `MongoPageSourceProvider`?